src/dev/code_coverage/ingest_coverage/ingest.js

```diff
@@ -7,75 +7,93 @@
+    log.verbose(`\n${ccMark} ${JSON.stringify(xs[0], null, 2)}`);
+  }
 };
 
-function doSend(index) {
-  return (redactedEsHostUrl) => (log) => (client) => async (payload) => {
-    const logF = logSend(true)(redactedEsHostUrl)(log);
-    await send(logF, index, redactedEsHostUrl, client, payload);
-  };
+function handleErrors(body, bulkResponse) {
+  return (log) =>
+    fromNullable(bulkResponse.errors) // check errors for null
+      .map(always(body)) // if errors is not null, pass the body to printErrors
+      .fold(id, parseErrors(log)(bulkResponse));
 }
 
-function dontSendButLog(log) {
-  return (payload) => {
-    logSend(false)(null)(log)(payload);
+function parseErrors(log) {
+  return (bulkResponse) => (innerBody) => {
+    const erroredDocuments = [];
+    // The items array has the same order of the dataset we just indexed.
+    // The presence of the `error` key indicates that the operation
+    // that we did for the document has failed.
+    bulkResponse.items.forEach((action, i) => {
+      const operation = Object.keys(action)[0];
+
+      if (action[operation].error) {
+        erroredDocuments.push({
+          // If the status is 429 it means that you can retry the document,
+          // otherwise it's very likely a mapping error, and you should
+          // fix the document before to try it again.
+          status: action[operation].status,
+          error: action[operation].error,
+          operation: innerBody[i * 2],
+          document: innerBody[i * 2 + 1],
+        });
+        log.error(`${ccMark} ${JSON.stringify(erroredDocuments, null, 2)}`);
+      }
+    });
   };
 }
 
-async function send(logF, idx, redactedEsHostUrl, client, requestBody) {
-  try {
-    await client.index(requestBody);
-    // logF(requestBody); // A simple way to speed things up, just log less output.
-  } catch (e) {
-    const { body } = requestBody;
-    const parsed = parse(body);
-    throw createFailError(errMsg(idx, redactedEsHostUrl, parsed, e));
-  }
+function parseIndexes(xs) {
+  return flatMap((doc) => {
+    const isTotal = !!doc.isTotal;
+    const _index = whichIndex(isResearchJob)(isTotal);
+    return [{ index: { _index } }, doc];
+  })(xs);
 }
 
-const sendMsg = (actuallySent, redactedEsHostUrl, payload) => {
-  const { index, body } = payload;
-  return `### ${actuallySent ? 'Sent' : 'Fake Sent'}:
-${redactedEsHostUrl ? `\t### ES Host: ${redactedEsHostUrl}` : ''}
-
-\t### Index: ${green(index)}
-
-\t### payload.body: ${body}
-`;
+export const ingestHelpers$ = (log) => async (dataStream) => {
+  fromNullable(process.env.NODE_ENV).fold(lazyF(ingestStream(log)(dataStream), id));
+
+  function ingestStream(log) {
+    return async (data$) => {
+      // See: https://www.elastic.co/guide/en/elasticsearch/client/javascript-api/current/client-helpers.html
+      const result = await client.helpers.bulk({
+        datasource: data$,
+        // flushInterval: 20000,
+        // concurrency: 6,
```

Comment:
is it potential args to use in future or test leftovers?

Reply:
Yeah I spoke to @LeeDr about this fn.  We decided to leave it in for now, but we can drop as it's currently unused. 
Actually, it's a fn that uses an experimental api that essentially does some of what our custom solution does.  As it's experimental, I wasn't comfortable using it as it could change: https://www.elastic.co/guide/en/elasticsearch/client/javascript-api/current/client-helpers.html

If you feel we should drop it, I've no issue with that as it's currently "dead code" for now.

Reply:
thanks for explanation, I don't mind keeping it as is